Kconfig.zephyr

```diff
@@ -50,6 +54,18 @@ source "subsys/Kconfig"
 
 osource "$(TOOLCHAIN_KCONFIG_DIR)/Kconfig"
 
+config SYSBUILD_CONTROLLED
+	bool
+	help
+	  This setting is set by sysbuild whenever sysbuild is used.
+	  It must never be selected by anything else than sysbuild but
+	  settings can rely on this to prevent users from configuring settings
+	  which are controlled by sysbuild, for example like:
+	  `bool "Foo" if !SYSBUILD_CONTROLLED`
+	  this line ensures that Foo can only be controlled through sysbuild
+	  when sysbuild is in use. When sysbuild is not in use, then users can
+	  control the setting.
```

Comment:
```suggestion
	  when sysbuild is in use. When sysbuild is not in use, then users can
	  freely change settings that would otherwise be controlled by sysbuild.
```